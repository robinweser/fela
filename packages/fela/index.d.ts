declare module "fela" {
  import * as CSS from 'csstype';
  import { TRuleType, TKeyframeType, TFontType, TStaticType, TClearType } from 'fela-utils';

  export type TRuleProps = {};
  export type TRule<T = TRuleProps> = (props: T, renderer: IRenderer) => IStyle
  export type TKeyFrame<T = TRuleProps> = (props: T, renderer: IRenderer) => {
    from?: IStyle,
    to?: IStyle,

    [persent: string]: IStyle | undefined;
  };

  type TRendererCreator = (config?: IConfig) => IRenderer;
  type TPlugin = (style: IStyle) => IStyle; //http://fela.js.org/docs/advanced/Plugins.html
  type TEnhancer = (renderer: IRenderer) => IRenderer; //http://fela.js.org/docs/advanced/Enhancers.html

  type TSubscribeMessageType = TRuleType | TKeyframeType | TFontType | TStaticType | TClearType

  interface ISubscribeMessage {
    type: TSubscribeMessageType;
  }
  interface ISubscribeRuleOrStaticObjectMessage extends ISubscribeMessage { static?: boolean; declaration: string; selector: string; media: string; }
  interface ISubscribeKeyframesMessage extends ISubscribeMessage { name: string; keyframe: string; }
  interface ISubscribeFontFaceMessage extends ISubscribeMessage { fontFamily: string; fontFace: string; }
  interface ISubscribeStaticStringMessage extends ISubscribeMessage { css: string; }
  interface ISubscribeClearMessage extends ISubscribeMessage { }

  interface IRenderer {
    renderRule<T = TRuleProps>(rule: TRule<T>, props: T): string
    renderKeyframe<T = TRuleProps>(keyFrame: TKeyFrame<T>, props: T): string;
    renderFont<T = TRuleProps>(family: string, files: Array<string>, props: T): void;
    renderStatic(style: string, selector?: string): void;
    renderStatic(style: IStyle, selector: string): void;
    renderToString(): string;
    subscribe(event: (msg: ISubscribeRuleOrStaticObjectMessage | ISubscribeKeyframesMessage | ISubscribeFontFaceMessage | ISubscribeStaticStringMessage | ISubscribeClearMessage) => void): { unsubscribe: () => void; }
    clear(): void;
  }

  //http://fela.js.org/docs/advanced/RendererConfiguration.html
  interface IConfig {
    plugins?: Array<TPlugin>;
    keyframePrefixes?: Array<string>;
    enhancers?: Array<TEnhancer>;
    mediaQueryOrder?: Array<string>;
    selectorPrefix?: string;
    specificityPrefix?: string;
    filterClassName?: (className: string) => boolean;
    devMode?: boolean;
  }

  export interface IStyle extends CSS.Properties<string | number> {
    // for selectors and pseudo classes use fela-plugin-typescript
  }

  function createRenderer(config?: IConfig): IRenderer;

  function combineRules<A, B>(a: TRule<A>, b: TRule<B>): TRule<A & B>
  function combineRules<A, B, C>(
    a: TRule<A>,
    b: TRule<B>,
    c: TRule<C>,
  ): TRule<A & B & C>
  function combineRules(...rules: Array<TRule>): TRule

  function enhance(...enhancers: Array<TEnhancer>): (rendererCreator: TRendererCreator) => TRendererCreator;
}

declare module "fela-dom" {
  import { IRenderer } from 'fela';

  function render(renderer: IRenderer): void;
  function rehydrate(renderer: IRenderer): void;
  function renderToMarkup(renderer: IRenderer): string;
  function renderToSheetList(renderer: IRenderer): {
    css: string,
    type: 'RULE' | 'KEYFRAME' | 'FONT' | 'STATIC',
    media?: string,
    support?: boolean,
  }[];
}

declare module "fela-tools" {
  import { TRule, TRuleProps, IStyle, IRenderer } from "fela";

  export type TMultiRuleObject<Props = TRuleProps, Styles = {}> = {[key in keyof Styles]: TRule<Props> | IStyle}
  export type TMultiRuleFunction<Props = TRuleProps, Styles = {}> = (props: Props, renderer: IRenderer) => TMultiRuleObject<Props, Styles>
  export type TMultiRule<Props = TRuleProps, Styles = {}> = TMultiRuleObject<Props, Styles> | TMultiRuleFunction<Props, Styles>

  export type TPartialMultiRuleObject<Props = TRuleProps, Styles = {}> = Partial<TMultiRuleObject<Props, Styles>>
  export type TPartialMultiRuleFunction<Props = TRuleProps, Styles = {}> = (props: Props, renderer: IRenderer) => TPartialMultiRuleObject<Props, Styles>
  export type TPartialMultiRule<Props = TRuleProps, Styles = {}> = TPartialMultiRuleObject<Props, Styles> | TPartialMultiRuleFunction<Props, Styles>

  export type TNormalizedMultiRule<Props = TRuleProps, Styles = {}> = (props: Props, renderer: IRenderer) => {[key in keyof Styles]: TRule<Props>}

  function combineMultiRules<A, SA, B, SB>(
    a: TMultiRule<A, SA>,
    b: TMultiRule<B, SB>
  ): TNormalizedMultiRule<A & B, SA & SB>
  function combineMultiRules<A, SA, B, SB, C, SC>(
    a: TMultiRule<A, SA>,
    b: TMultiRule<B, SB>,
    c: TMultiRule<C, SC>,
  ): TNormalizedMultiRule<A & B & C, SA & SB & SC>
  function combineMultiRules(...rules: Array<TMultiRule>): TNormalizedMultiRule

  function mapValueToMediaQuery(
    queryValueMap: { [key: string]: string },
    mapper: ((value: string) => object) | string
  ): object;

  function renderToElement(
    renderer: IRenderer,
    mountNode: { textContent: string },
  ): (() => void);

  function renderToString(
    renderer: IRenderer,
  ): string;
}

/**
 * ENHANCERS
 */
declare module "fela-beautifier" {
  import { TEnhancer } from "fela";

  export default function(options?: object): TEnhancer;
}

declare module "fela-identifier" {
  import { TRule, TEnhancer } from "fela";

  interface Configs {
    prefix?: string;
    generator?: (name?: string, index?: number) => string;
  }

  type Identifier = (name?: string) => TRule & {
    className: string;
    toString(): string;
  };

  export default function(configs?: Configs): TEnhancer & Identifier;
}

declare module "fela-layout-debugger" {
  import { TEnhancer } from "fela";

  interface DebuggerOptions {
    mode?: "outline" | "backgroundColor";
    thickness?: number;
  }

  export default function(options?: DebuggerOptions): TEnhancer;
}

declare module "fela-logger" {
  import { TEnhancer } from "fela";

  interface LoggerOptions {
    logCSS?: boolean;
    formatCSS?: boolean;
  }

  export default function(options?: LoggerOptions): TEnhancer;
}

declare module "fela-monolithic" {
  import { TEnhancer } from "fela";

  interface MonolithicOptions {
    prettySelectors?: boolean;
  }

  export default function(options?: MonolithicOptions): TEnhancer;
}

declare module "fela-perf" {
  import { TEnhancer } from "fela";

  export default function(): TEnhancer;
}

declare module "fela-statistics" {
  import { TEnhancer } from "fela";

  export default function(): TEnhancer;
}

declare module "fela-utils" {
  export type TRuleType = "RULE"
  export type TKeyframeType = "KEYFRAME"
  export type TFontType = "FONT"
  export type TStaticType = "STATIC"
  export type TClearType = "CLEAR"

  export const RULE_TYPE: TRuleType
  export const KEYFRAME_TYPE: TKeyframeType
  export const FONT_TYPE: TFontType
  export const STATIC_TYPE: TStaticType
  export const CLEAR_TYPE: TClearType
}

declare module "fela-sort-media-query-mobile-first" {
  import { TEnhancer } from "fela";

  export default function(): TEnhancer;
}

/**
 * PLUGINS
 */
declare module "fela-plugin-bidi" {
  import { TPlugin } from "fela";

  export default function(flowDirection: 'ltr' | 'rtl'): TPlugin;
}

declare module "fela-plugin-custom-property" {
  import { TPlugin } from "fela";

  interface CustomProperties {
    [property: string]: (value: any) => any,
  }

  export default function(properties: CustomProperties): TPlugin;
}

declare module "fela-plugin-embedded" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-extend" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-fallback-value" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-friendly-pseudo-class" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-important" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-isolation" {
  import { TPlugin } from "fela";

  interface IsolationOptions {
    exclude?: string[];
  }

  export default function(options?: IsolationOptions): TPlugin;
}

declare module "fela-plugin-logger" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-named-keys" {
  import { TPlugin } from "fela";

  interface MediaQueryMap {
    [key: string]: string;
  }

  export default function(mediaQueryMap: MediaQueryMap): TPlugin;
}

declare module "fela-plugin-native-media-query" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-placeholder-prefixer" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-fullscreen-prefixer" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-pseudo-prefixer" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-theme-value" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-prefixer" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-rtl" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-simulate" {
  import { TPlugin } from "fela";

  export default function(): TPlugin;
}

declare module "fela-plugin-unit" {
  import { TPlugin } from "fela";

  export type Unit = "ch" | "em" | "ex" | "rem" | "vh" | "vw" | "vmin" | "vmax" | "px" | "cm" | "mm" | "in" | "pc" | "pt" | "mozmm";

  export interface UnitPerProperty {
    [key: string]: string;
  }

  export default function(
    unit?: Unit,
    unitPerProperty?: UnitPerProperty,
    isUnitlessProperty?: (property: string) => boolean,
  ): TPlugin;
}

declare module "fela-plugin-validator" {
  import { TPlugin } from "fela";

  interface Configs {
    logInvalid?: boolean;
    deleteInvalid?: boolean;
    useCSSLint?: boolean | object;
  }

  export default function(configs?: Configs): TPlugin;
}

/**
 * PRESETS
 */
declare module "fela-preset-web" {
  import { TPlugin } from "fela";
  import { Unit, UnitPerProperty } from "fela-plugin-unit";

  type UnitConfig1 = [Unit]
  type UnitConfig2 = [Unit, UnitPerProperty]
  type UnitConfig3 = [Unit, UnitPerProperty, (property: string) => boolean]

  type UnitConfig = UnitConfig1 | UnitConfig2 | UnitConfig3

  export function createWebPreset({ unit }: { unit?: UnitConfig }): TPlugin[];

  const presets: TPlugin[];
  export default presets;
}

declare module "fela-preset-dev" {
  import { TPlugin } from "fela";

  const presets: TPlugin[];
  export default presets;
}
