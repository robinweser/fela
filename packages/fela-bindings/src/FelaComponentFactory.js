/* @flow */
import { combineRules } from 'fela'

export default function FelaComponentFactory(
  BaseComponent: any,
  createElement: Function,
  FelaTheme: Function,
  contextTypes?: Object
): Function {
  function FelaComponent(
    { render, rule, extend, ...otherProps },
    { renderer }
  ) {
    const combinedRule = extend ? combineRules(rule, extend) : rule

    return createElement(FelaTheme, {
      render: theme =>
        render({
          className: renderer.renderRule(combinedRule, {
            ...otherProps,
            theme,
          }),
          theme,
        }),
    })
  }

  if (contextTypes) {
    FelaComponent.contextTypes = contextTypes
  }

  return FelaComponent
}
