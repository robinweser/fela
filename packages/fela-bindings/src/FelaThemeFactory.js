/* @flow */
import { THEME_CHANNEL } from './themeChannel'

export default function FelaThemeFactory(
  BaseComponent: any,
  contextTypes?: Object
): any {
  class FelaTheme extends BaseComponent {
    unsubscribe: ?Function
    constructor(props, context) {
      super(props, context)

      this.state = {
        theme: context[THEME_CHANNEL] ? context[THEME_CHANNEL].get() : {},
      }
    }

    componentWillMount() {
      if (this.context[THEME_CHANNEL]) {
        this.unsubscribe = this.context[THEME_CHANNEL].subscribe(properties =>
          this.setState({
            theme: properties,
          })
        )
      }
    }

    componentWillUnmount() {
      if (this.unsubscribe) {
        this.unsubscribe()
      }
    }

    render() {
      return this.props.render(this.state.theme)
    }
  }

  if (contextTypes) {
    FelaTheme.contextTypes = contextTypes
  }

  return FelaTheme
}
