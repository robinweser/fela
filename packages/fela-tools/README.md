# fela-tools

A set of useful tools to work with Fela.

<img alt="npm version" src="https://badge.fury.io/js/fela-tools.svg"> <img alt="npm downloads" src="https://img.shields.io/npm/dm/fela-tools.svg"> <a href="https://bundlephobia.com/result?p=fela-tools@latest"><img alt="Bundlephobia" src="https://img.shields.io/bundlephobia/minzip/fela-tools.svg"></a>

## Installation
```sh
yarn add fela-tools
```
You may alternatively use `npm i --save fela-tools`.


## Usage
The API is exported by the `fela-tools` package by default e.g.

```javascript
import { StyleSheet } from 'fela-tools'
```

## API Reference

* [`mapValueToMediaQuery(queryValueMap, mapper)`](docs/mapValueToMediaQuery.md)
* [`renderToString(renderer)`](docs/renderToString.md)
* [StyleSheet](docs/StyleSheet.md)

## License
Fela is licensed under the [MIT License](http://opensource.org/licenses/MIT).<br>
Documentation is licensed under [Creative Commons License](http://creativecommons.org/licenses/by/4.0/).<br>
Created with â™¥ by [@robinweser](http://weser.io) and all the great contributors.
