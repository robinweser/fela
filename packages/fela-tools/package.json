{
  "name": "fela-tools",
  "version": "11.2.0",
  "description": "Useful tools for working with Fela",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "sideEffects": false,
  "files": [
    "LICENSE",
    "README.md",
    "lib/**",
    "es/**"
  ],
  "repository": "https://github.com/robinweser/fela/",
  "keywords": [
    "fela",
    "fela-tool",
    "cssinjs"
  ],
  "author": "Robin Frischmann",
  "license": "MIT",
  "peerDependencies": {
    "fela": "^11.2.0"
  },
  "dependencies": {
    "css-in-js-utils": "^3.0.0",
    "fast-loops": "^1.0.0",
    "fela": "^11.2.0",
    "fela-utils": "^11.2.0"
  },
  "gitHead": "e1a48f5d72623a0f95084dc07998dd85aa175bd1"
}
