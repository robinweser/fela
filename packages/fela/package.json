{
  "name": "fela",
  "version": "11.4.0",
  "description": "State-Driven Styling in JavaScript",
  "typings": "index.d.ts",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "sideEffects": false,
  "scripts": {
    "build:umd": "webpack --config webpack.config.js"
  },
  "files": [
    "LICENSE",
    "README.md",
    "index.d.ts",
    "lib/**",
    "es/**",
    "umd/**"
  ],
  "repository": "https://github.com/robinweser/fela/",
  "keywords": [
    "fela",
    "dynamic styling",
    "stylesheet",
    "css",
    "styling",
    "functional styling",
    "functional css",
    "atomic css",
    "state-driven",
    "atomic design",
    "state-driven styling",
    "cssinjs"
  ],
  "author": "Robin Weser",
  "license": "MIT",
  "dependencies": {
    "css-in-js-utils": "^3.0.0",
    "csstype": "^2.5.5",
    "fast-loops": "^1.0.0",
    "fela-utils": "^11.4.0",
    "isobject": "^3.0.1"
  },
  "gitHead": "4cc4686d9e16877251ac5c73e161076ec7c113a1"
}
