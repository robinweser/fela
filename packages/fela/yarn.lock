# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
# yarn lockfile v1


css-in-js-utils@2.0.0, css-in-js-utils@^2.0.0:
  version "2.0.0"
  resolved "https://registry.yarnpkg.com/css-in-js-utils/-/css-in-js-utils-2.0.0.tgz#5af1dd70f4b06b331f48d22a3d86e0786c0b9435"
  dependencies:
    hyphenate-style-name "^1.0.2"

fast-loops@^1.0.0:
  version "1.0.0"
  resolved "https://registry.yarnpkg.com/fast-loops/-/fast-loops-1.0.0.tgz#2cdd7e0ff67343b2b5f5e627d855a50b4bed559a"

fela-tools@^5.1.2:
  version "5.1.2"
  resolved "https://registry.yarnpkg.com/fela-tools/-/fela-tools-5.1.2.tgz#c6760b42d72116911c2c0341b6d880cfc11c76a8"
  dependencies:
    css-in-js-utils "2.0.0"
    fast-loops "^1.0.0"
    fela "^6.1.3"
    fela-utils "^8.0.3"

fela-utils@^8.0.3:
  version "8.0.3"
  resolved "https://registry.yarnpkg.com/fela-utils/-/fela-utils-8.0.3.tgz#4a1728e7f5711e26f6f37da9c85a6b270c84c466"
  dependencies:
    css-in-js-utils "^2.0.0"
    fast-loops "^1.0.0"

fela@^6.1.3:
  version "6.1.3"
  resolved "https://registry.yarnpkg.com/fela/-/fela-6.1.3.tgz#c3bc8726dada63c23914bed9ca79d902e61e9a74"
  dependencies:
    css-in-js-utils "2.0.0"
    fast-loops "^1.0.0"
    fela-utils "^8.0.3"

hyphenate-style-name@^1.0.2:
  version "1.0.2"
  resolved "https://registry.yarnpkg.com/hyphenate-style-name/-/hyphenate-style-name-1.0.2.tgz#31160a36930adaf1fc04c6074f7eb41465d4ec4b"

string-hash@^1.1.3:
  version "1.1.3"
  resolved "https://registry.yarnpkg.com/string-hash/-/string-hash-1.1.3.tgz#e8aafc0ac1855b4666929ed7dd1275df5d6c811b"
