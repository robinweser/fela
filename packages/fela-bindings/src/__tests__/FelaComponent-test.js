import React, { Component, createElement } from 'react'
import PropTypes from 'prop-types'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { css } from 'js-beautify'

import { renderToString } from 'fela-tools'
import { createRenderer } from 'fela'

import FelaThemeFactory from '../FelaThemeFactory'
import FelaComponentFactory from '../FelaComponentFactory'
import createTheme from '../createTheme'
import { THEME_CHANNEL } from '../themeChannel'

const FelaTheme = FelaThemeFactory(Component, {
  [THEME_CHANNEL]: PropTypes.object,
})

const FelaComponent = FelaComponentFactory(
  Component,
  createElement,
  FelaTheme,
  {
    [THEME_CHANNEL]: PropTypes.object,
    renderer: PropTypes.object,
  }
)
describe('Using the FelaComponent component', () => {
  it('correctly render a fela rule', () => {
    const rule = ({ color }) => ({
      fontSize: '12px',
      color,
    })

    const renderer = createRenderer()

    const wrapper = mount(
      <FelaComponent
        rule={rule}
        color="red"
        render={({ className }) => (
          <div className={className}>I am red and written in 12px.</div>
        )}
      />,
      {
        context: {
          renderer,
        },
      }
    )

    expect([css(renderToString(renderer)), toJson(wrapper)]).toMatchSnapshot()
  })

  it('correctly pass the theme', () => {
    const rule = ({ color, theme }) => ({
      fontSize: theme.fontSize,
      color,
    })

    const themeContext = createTheme({
      fontSize: '15px',
    })

    const renderer = createRenderer()

    const wrapper = mount(
      <FelaComponent
        rule={rule}
        color="red"
        render={({ className, theme }) => (
          <div className={className}>
            I am red and written in {theme.fontSize}.
          </div>
        )}
      />,
      {
        context: {
          [THEME_CHANNEL]: themeContext,
          renderer,
        },
      }
    )

    expect([css(renderToString(renderer)), toJson(wrapper)]).toMatchSnapshot()
  })

  it('correctly combine extending rules', () => {
    const rule = ({ color }) => ({
      fontSize: '12px',
      color,
    })

    const extend = ({ fontSize }) => ({
      fontSize,
      backgroundColor: 'blue',
    })

    const renderer = createRenderer()

    const wrapper = mount(
      <FelaComponent
        rule={rule}
        extend={extend}
        color="red"
        fontSize="15px"
        render={({ className }) => (
          <div className={className}>I am red on blue and written in 15px.</div>
        )}
      />,
      {
        context: {
          renderer,
        },
      }
    )

    expect([css(renderToString(renderer)), toJson(wrapper)]).toMatchSnapshot()
  })
})
