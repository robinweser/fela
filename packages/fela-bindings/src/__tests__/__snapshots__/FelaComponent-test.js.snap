// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using the FelaComponent component correctly combine extending rules 1`] = `
Array [
  ".a {
    font-size: 15px
}

.b {
    color: red
}

.c {
    background-color: blue
}",
  <FelaComponent
    color="red"
    extend={[Function]}
    fontSize="15px"
    render={[Function]}
    rule={[Function]}
>
    <FelaTheme
        render={[Function]}
    >
        <div
            className="a b c"
        >
            I am red on blue and written in 15px.
        </div>
    </FelaTheme>
</FelaComponent>,
]
`;

exports[`Using the FelaComponent component correctly pass the theme 1`] = `
Array [
  ".a {
    font-size: 15px
}

.b {
    color: red
}",
  <FelaComponent
    color="red"
    render={[Function]}
    rule={[Function]}
>
    <FelaTheme
        render={[Function]}
    >
        <div
            className="a b"
        >
            I am red and written in 
            15px
            .
        </div>
    </FelaTheme>
</FelaComponent>,
]
`;

exports[`Using the FelaComponent component correctly render a fela rule 1`] = `
Array [
  ".a {
    font-size: 12px
}

.b {
    color: red
}",
  <FelaComponent
    color="red"
    render={[Function]}
    rule={[Function]}
>
    <FelaTheme
        render={[Function]}
    >
        <div
            className="a b"
        >
            I am red and written in 12px.
        </div>
    </FelaTheme>
</FelaComponent>,
]
`;
